### ISSUE TITLE
WSL support like Quokka?

### ISSUE THREAD

**Issue by rishispeets** [wallabyjs/public#1922]
You added WSL support in one of the recent updates for Quokka. Thanks for that! Are you planning on doing the same for Wallaby?

**Comment by smcenlly** [comment]
It wasn't on our list of short term plans, but we will look at adding support over the next 1-2 months (if not sooner).

**Comment by dave-irvine** [comment]
Definitely something that would help me out a lot as I can't even figure a workaround using `wallaby.node` configuration.

**Comment by NikGovorov** [comment]
Hi Guys,

The feature is implemented in Wallaby.js for VS Code extension 1.0.109. Please find more information in the [documentation](https://wallabyjs.com/docs/intro/get-started-vscode.html#windows-subsystem-for-linux-wsl).

**Comment by dave-irvine** [comment]
@NikGovorov no luck unfortunately, updated the extension, have set

```{
    "wallaby.useWsl": true
}```

in my VSCode settings, restarted the editor.

When I try to Start wallaby I get:

```
Can not start node.js process (node), make sure your system has node.js installed.
```

In my WSL shell I can ask where node is:

```
08:28 $ which node
/usr/bin/node
```

If I try to use `wallaby.node` configuration setting to set this location, and restart the IDE, the new error is:

```Can not start node.js process (/usr/bin/node), make sure your system has node.js installed.```

**Comment by ArtemGovorov** [comment]
Looks like maybe `/usr/bin/node` is a symlink, but `wallaby.node` needs an absolute path to node binary. Try

- run `ls -la /usr/bin | grep "\->"`,
- find where `node` points, 
- copy its absolute path (ie. no `../`, full absolute path) to the `wallaby.node` setting,
- restart your editor.

**Comment by dave-irvine** [comment]
@ArtemGovorov it's a binary, not a symlink, sorry :(

```
10:13 $ ls -la /usr/bin | grep node
...snip
-rwxr-xr-x 1 root   root       18369616 Mar 23  2018 node
...snip
```

**Comment by NikGovorov** [comment]
Could you please in command prompt run ```wsl node -v``` and paste the results?

**Comment by dave-irvine** [comment]
```
C:\Users\dave>wsl node -v
v8.10.0
```

**Comment by NikGovorov** [comment]
Thanks, could you please run ```type "%USERPROFILE%\.vscode\extensions\wallabyjs.wallaby-vscode-1.0.109\wallaby\package.json"``` in command prompt and publish the results?

**Comment by dave-irvine** [comment]
```
{
  "name": "Wallaby.js",
  "version": "1.0.620",
  "author": "Wallaby.js",
  "license": "Proprietary",
  "homepage": "http://wallabyjs.com",
  "dependencies": {
    "acorn": "6.0.2",
    "acorn-jsx": "5.0.0",
    "chokidar": "1.0.3",
    "chrome-launcher": "0.8.0",
    "chrome-remote-interface": "0.24.4",
    "coffee-script": "1.10.0",
    "commander": "2.5.0",
    "compression": "1.7.2",
    "debug": "2.1.0",
    "escodegen-wallaby": "1.6.20",
    "express": "4.11.2",
    "fs-extra": "0.12.0",
    "graceful-fs": "4.1.3",
    "lodash": "3.0.0",
    "lru-cache": "2.5.0",
    "minimatch": "3.0.4",
    "node-uuid": "1.4.3",
    "phantom": "0.7.2",
    "q": "1.5.0",
    "semver": "4.3.3",
    "source-map": "0.4.3",
    "typescript": "2.5.2",
    "underscore.string": "2.3.3",
    "ws": "1.1.1"
  },
  "devDependencies": {
    "browserify-transform-tools": "1.5.0",
    "escodegen": "1.6.1",
    "expect.js": "0.3.1",
    "grunt": "*",
    "grunt-browserify": "*",
    "grunt-bump": "0.0.16",
    "grunt-contrib-clean": "*",
    "grunt-contrib-compress": "0.12.0",
    "grunt-contrib-copy": "*",
    "grunt-contrib-uglify": "*",
    "grunt-ts": "^6.0.0-beta.16",
    "npm": "4.1.2",
    "ts-node": "^3.1.0"
  },
  "private": true
}
```

**Comment by NikGovorov** [comment]
Unfortunately I'm not sure if the version of Core that is covered by your license and you have installed supports WSL. Could you please try to reset your license?

1. Close VS Code
2. Run ```rename "%USERPROFILE%\.wallaby\key.lic" "expired_key.lic"```
3. Run ```rmdir /s /q "%USERPROFILE%\.vscode\extensions\wallabyjs.wallaby-vscode-1.0.109\wallaby"```
4. Start VS Code
5. Start Wallaby.js - it should start in trial mode with the latest Core

To get back to version 1.0.620 which is covered by your license, please follow the steps: 

1. Close VS Code
2. Run ```rename "%USERPROFILE%\.wallaby\expired_key.lic" "key.lic"```
3. Run ```rmdir /s /q "%USERPROFILE%\.vscode\extensions\wallabyjs.wallaby-vscode-1.0.109\wallaby"```
4. Start VS Code
5. Start Wallaby.js

**Comment by dave-irvine** [comment]
That has resolved the issue :) I've renewed my license.

Thanks!

**Comment by remshams** [comment]
Are there any plans to also provide WSL Support for the JetBrains IDEs?

**Comment by smcenlly** [comment]
We have supported WSL for JetBrains IDEs since the end of April 2019. Check out this medium blog post for more information: https://medium.com/@wallabyjs/new-wallaby-js-for-jetbrains-editors-496da2c30d34

**Comment by remshams** [comment]
Thank you so much for the response. Works perfectly now.
I tried to figure it out half a year okay but obviously the "Use WSL" option was not yet available at that time.

I could not find anything in the docs related to the Jetbrain IDE's in regards to WSL support (for VSCode there is a dedicated section). That's why I wrote the comment here.
Would it make sense to add it here: https://wallabyjs.com/docs/intro/get-started-jetbrains.html?

Thanks again for the speedy answer/support.
